documentation/release_notes.markdown

```diff
@@ -10,20 +10,46 @@ canonical: "/puppetserver/latest/release_notes.html"
 [puppetserver.conf]: ./config_file_puppetserver.markdown
 [product.conf]: ./config_file_product.markdown
 
-For release notes on versions of Puppet Server prior to Puppet Server 5, see [docs.puppet.com](https://docs.puppet.com/puppetserver/2.7/release_notes.html).
+For release notes on versions of Puppet Server prior to Puppet Server 5, see [docs.puppet.com](https://docs.puppet.com/puppetserver/2.8/release_notes.html).
 
 ## Puppet Server 5.1.0
 
 Released September 12, 2017
 
-This is a feature release of Puppet Server.
+This is a feature and bug-fix release of Puppet Server.
 
 ### New Feature: Automatic CRL refresh on certificate revocation
 
-Puppet Server 5.1.0 includes the ability to automatically refresh the CRL in the running SSL context when any changes to that file have occurred, namely the addition of a revoked certificate. Prior to this release, revoking an agent's certificate required restarting the Puppet Server process before that revocation would be honored and the agent denied authentication. Now revocation will be effective psuedo-immediately (some threshold of milliseconds) without requiring a restart of the server.
+Puppet Server 5.1.0 includes the ability to automatically refresh the certificate revocation list (CRL) in the running SSL context when any changes to that file have occurred, namely the addition of a revoked certificate. Prior to this release, revoking an agent's certificate required restarting the Puppet Server process before that revocation would be honored and the agent denied authentication. Revocation is now effective within milliseconds and does not require restarting server.
 
 -   [SERVER-1933](https://tickets.puppetlabs.com/browse/SERVER-1933) / [TK-451](https://tickets.puppetlabs.com/browse/TK-451)
 
+### New Feature: Puppet agents retry requests on a configurable delay if Puppet Server is busy
+
+When a group of Puppet agents start their Puppet runs together, they can form a "thundering herd" capable of exceeding Puppet Server's available resources. This results in a growing backlog of requests from Puppet agents waiting for a JRuby instance to become free before their request can be processed. If this backlog exceeds the size of the Server's Jetty thread pool, other requests (such as status checks) start timing out. (For more information about JRubies and Server performance, see [Applying metrics to improve performance](https://docs.puppet.com/puppetserver/latest/puppet_server_metrics_performance.html#measuring-capacity-with-jrubies).)
+
+In previous versions of Puppet Server, administrators had to manually remediate this situation by separating groups of agent requests, for instance through rolling restarts. In Server 5.1.0, administrators can optionally have Server return a 503 response containing a `Retry-After` header to requests when the JRuby backlog exceeds a certain limit, causing agents to pause before retrying the request.
```

Comment:
IIRC, the header is now sent by the server but the agent code isn't merged yet. Looks like there's a lot of movement on it, do we know for sure it will ship?

Reply:
I don't know. Would have to ask on the Agent side. Due to the scheduling wackiness for these specific docs PRs (I'm on PTO and off grid for the release date), I'll remove this and we can add it back in post-release.